types/bluebird/index.d.ts

```diff
@@ -307,7 +318,7 @@ declare class Bluebird<R> implements PromiseLike<R>, Bluebird.Inspection<R> {
    * });
    * </code>
    */
-  call(propertyName: keyof R, ...args: any[]): Bluebird<any>;
+  call<P extends keyof R, U extends ExtractFunction<R[P]>>(propertyName: P, ...args: any[]): Bluebird<U extends never ? never : ReturnType<U>>;
```

Comment:
And since `Extract<T, Function>` is the same length as the type alias itself you could IMHO just remove the `ExtractFunction` alias and write `U extends Extract<R[P], Function>` here instead.

Reply:
This review will apply when applying the above review.